test/upgrade/test-upgrade.bats

```diff
@@ -239,19 +260,26 @@ failed    | exited     | 17
 }
 
 @test "network - curl" {
-    run curl --max-time 3 -s 127.0.0.1:$HOST_PORT/index.txt
+    run -0 curl --max-time 3 -s 127.0.0.1:$HOST_PORT/index.txt
     is "$output" "$RANDOM_STRING_1" "curl on running container"
 }
 
 # IMPORTANT: connect should happen before restart, we want to check
 # if we can connect on an existing running container
+#
+# FIXME FIXME FIXME 2024-02: does this even make sense in 2024?
+# There seem to be some iptables commands involved, and I don't see
+# how those can possibly run in containerized podman?
 @test "network - connect" {
-    skip_if_version_older 2.2.0
-    touch $PODMAN_UPGRADE_WORKDIR/ran-network-connect-test
+    # FIXME: for debugging
+    run_podman network ls
 
-    run_podman network connect mynetwork myrunningcontainer
+    run_podman network connect $MYTESTNETWORK myrunningcontainer
+    # FIXME: this fails with "netavark: iptables: No * by that name"
+    # ...and if we comment it out, curl fails (28, timeout), presumably
+    # because of the above network connect
     run_podman network disconnect podman myrunningcontainer
-    run curl --max-time 3 -s 127.0.0.1:$HOST_PORT/index.txt
+    run -0 curl --max-time 3 -s 127.0.0.1:$HOST_PORT/index.txt
     is "$output" "$RANDOM_STRING_1" "curl on container with second network connected"
```

Comment:
help

Reply:
CI seems to pass on that, did you mix CNI and netavark on your system again?
Or maybe mixing between iptables-legacy and iptables-nft?

Reply:
CNI was my first thought, of course. I'm 99% sure that's not it because there's no `podmanX` interface. It could be iptables? These are the packages on my laptop:
```console
$ rpm -qa|sed -ne 's/^iptables-//' -e 's/-1.8.9-5.*$//p'|sort|fmt
compat legacy legacy-libs libs nft utils
```
So, there's compat and legacy and nft. Presumably because of my long upgrade path. Is that not a supported configuration, and what should I do to fix it?

Reply:
Having both installed is fine, usually you have to switch via alternatives between them. Both should work with netavark.
You can just run `iptables -V` to check which one is active. What I was hinting at is that we first run in the old podman container with a potential different iptables, so if container has iptables-legacy while the host uses iptables-nft then an error with no rule/chain could make sense because they use different kernel backends and cannot see the rules from each other.

My understanding is that they should never be mixed on a running system as different rules can cause weird conflicts, so if one switches the iptables backend it is best to reboot first.

Reply:
yes the image uses legacy so this cannot work correctly in this combination
```
$ podman run --rm quay.io/podman/stable:v4.1.0 iptables -V
iptables v1.8.7 (legacy)
```
And it looks like the image has only legacy installed so the only way to run these tests successfully is to use iptables-legacy on the host or patch up the podman image to include and use iptables-nft but that seems harder.